easybuild/tools/run.py

```diff
@@ -80,16 +80,18 @@ def run_cmd_cache(func):
     @functools.wraps(func)
     def cache_aware_func(cmd, *args, **kwargs):
         """Retrieve cached result of selected commands, or run specified and collect & cache result."""
-
-        # fetch from cache if available, cache it if it's not
-        if cmd in cache:
-            _log.debug("Using cached value for command '%s': %s", cmd, cache[cmd])
-            return cache[cmd]
+        # fetch from cache if available, cache it if it's not, but only on cmd strings
```

Comment:
You only cache based on strings. It seems safer to harden the code for what is expected rather than trying to handle all sorts of things that aren't necessary (including lists since none of the cached commands are lists at the moment and it doesn't work).